Documentation/building/dev-workflow.md

```diff
@@ -0,0 +1,46 @@
+Dev Workflow
+===============
+
+The dev workflow describes the development process to follow when working in our repo. 
+It is divided into specific tasks that are fast, transparent and easy to understand.
+
+This provides flexibility as with discrete tasks people can iterate on a specific one without needing to run the full workflow, other processes can reuse the tasks as needed and anyone can put together the tasks as wanted to create a customized workflow.
+
+## Process 
+![Dev Workflow process](../images/dev-workflow.png)
+
+All the tasks (except enlist, code, submit) will have scripts located at the root of the repo, and will have usage help to better understand what they provide.
+In addition, we want to have an “easy button” script for running many or all of these tasks sequentially in a single command.
+
+## Tasks
+
+**Clean**
+
+Cleans the local enlistment/environment, for example removing files not in the repository. 
+
+**Sync**
+
+The sync task gets the latest source history and packages in order to eliminate all the network traffic when we build. This way we are able to hit the network only when we are intentional about doing it and then be able to build even in offline mode. 
```

Comment:
> The sync task gets the latest source history and packages in order to eliminate all the network traffic when we build. This way we are able to hit the network only when we are intentional about doing it and then be able to build even in offline mode.

I'd really like to understand why we think "Airplane" mode needs to the the default.  I agree it should totally be possible to pre-cache everything I'd need from the build and then go offline, and that this makes sense for some workflows but I don't understand why we force the workflow by default.  Most of the time I have an internet connection. 

When I have to work internally in our TFS copy of CoreFX, I'm continually bitten by this, because I neglect to run "sync.cmd" and instead use the source control commands I'm used to.

I really feel like disabling restore during build is going to negatively impact productivity by forcing me to remember to invoke `sync.cmd` any time I touch a project.json or face obtuse errors.

This seems so foreign compared to every other project I've seen, why is it so important that we make it the default?